QGIS Plugin/templateselectordialog.py

```diff
@@ -340,75 +352,67 @@ def getCopyrightText(self):
         return copyrightText
 
     def openTemplate(self):
-
-        templateFilePath = self.getTemplateFilePath()
-
-        if not os.path.isfile(templateFilePath):
-            QtGui.QMessageBox.critical(self.iface.mainWindow(), \
-                                       'Template Not Found', \
-                                       'The requested template (%s) is not currently available.' \
-                                       % templateFilePath)
+        template_path = self.getTemplateFilePath()
+        if not os.path.isfile(template_path):
+            msg = 'The requested template {} is not currently available.'.format(template_path)
+            QMessageBox.critical(self.iface.mainWindow(), 'Template Not Found', msg)
             return
 
-        # Load replaceable text
-        projectTitle = self.ui.titleLineEdit.text()
-        projectSubTitle = self.ui.subtitleLineEdit.text()
-
-        # Set copyright
+        # Create a new print layout with name equal to the project title
+        project = QgsProject.instance()
+        layout_manager = project.layoutManager()
+        print_layout = QgsPrintLayout(project)
+        print_layout.setName(self.ui.titleLineEdit.text())
+        layout_manager.addLayout(print_layout)
 
-        copyrightIndex = self.ui.copyrightComboBox.currentIndex()
-        self.replaceMap['copyright'] = self.getCopyrightText()
-        self.replaceMap['title'] = projectTitle
-        self.replaceMap['subtitle'] = projectSubTitle
-
-        # Create a new Composer View with name equal to the project
-        # title
-        composerView = self.iface.createNewComposer(projectTitle)
-
-        # Load the template file, replacing any text we find in
-        # '[' and ']'
+        # Load the template file, replacing any text we find in '[' and ']'
         try:
-            tree = ET.parse(templateFilePath)
+            tree = ET.parse(template_path)
             if self.hasGridRefs():
                 # Replace the content of the 'gridref' element based on the selected POI layer
-                composerLabelElem = tree.find("//ComposerLabel/ComposerItem[@id='gridref']/..")
+                composerLabelElem = tree.find("//LayoutItem[@id='gridref']")
```

Comment:
Doing changes in via `ET.tree` was done as there were limitations with editing compositions with pyqgis. Now with Layouts they are gone, so you can (and should) get rid of that ugly XML editing. Please repleace it with QgsLayout methods like `itemById` `setText` etc.
( https://qgis.org/api/classQgsLayout.html )